src/helpers/figures/charts/runtime/chart_data_extractor.ts

```diff
@@ -44,11 +44,27 @@ import { recomputeZones } from "../../../recompute_zones";
 export function getBarChartData(
   definition: GenericDefinition<BarChartDefinition>,
   dataSets: DataSet[],
-  labelRange: Range | undefined,
+  labelRange: Range[] | undefined,
   getters: Getters
 ): ChartRuntimeGenerationArgs {
-  const labelValues = getChartLabelValues(getters, dataSets, labelRange);
-  let labels = labelValues.formattedValues;
+  let labels: string[] = [];
+  if (labelRange && labelRange.length) {
+    for (const range of labelRange) {
+      const labelValues = getChartLabelValues(getters, dataSets, range);
+      const values = labelValues.formattedValues;
+      if (!labels.length) {
+        labels = values;
+      } else {
+        for (let i = 0; i < values.length; i++) {
+          if (values[i]) {
+            labels[i] += " \n " + values[i];
```

Comment:
Didn't you tell me that we do'nt need to do "hacky" \n and that chartJS accepts multiple labels ranges ? OR did I misunderstood ?

Reply:
Yeah, this is done to keep a "classical" string as label here, to be able to proceed the label shift and all the filtering stuff without too many changes. And then we resplit it in an array (your second comment)